lib/ansible/modules/cloud/amazon/polly_speech.py

```diff
@@ -0,0 +1,211 @@
+    ]
+
+    module = AnsibleAWSModule(
+        argument_spec={
+            'lexicons': dict(type='list'),
+            'output_format': dict(type='str', choices=['json', 'mp3', 'ogg_vorbis', 'pcm'], required=True),
+            'sample_rate': dict(type='str', default='16000'),
+            'speech_mark_types': dict(type='list', choices=['sentence', 'ssml', 'viseme', 'word']),
+            'text_type': dict(type='str', choices=['ssml', 'text'], default='text'),
+            'voice_id': dict(
+                type='str',
+                choices=[
+                    'Geraint', 'Gwyneth', 'Mads', 'Naja', 'Hans', 'Marlene', 'Nicole', 'Russell', 'Amy', 'Brian',
+                    'Emma', 'Raveena', 'Ivy', 'Joanna', 'Joey', 'Justin', 'Kendra', 'Kimberly', 'Matthew', 'Salli', 'Conchita',
+                    'Enrique', 'Miguel', 'Penelope', 'Chantal', 'Celine', 'Mathieu', 'Dora', 'Karl', 'Carla', 'Giorgio',
+                    'Mizuki', 'Liv', 'Lotte', 'Ruben', 'Ewa', 'Jacek', 'Jan', 'Maja', 'Ricardo', 'Vitoria', 'Cristiano',
+                    'Ines', 'Carmen', 'Maxim', 'Tatyana', 'Astrid', 'Filiz', 'Vicki', 'Takumi', 'Seoyeon', 'Aditi'
+                ],
+                required=True
+            ),
+            'source': dict(type='str', choices=['local', 's3'], required=True),
+            'src_bucket': dict(type='str'),
+            'src_path': dict(type='str'),
+            'destination': dict(type='str', choices=['local', 's3'], required=True),
+            'dst_bucket': dict(type='str'),
+            'dst_path': dict(type='str')
+        },
+        supports_check_mode=True,
+        required_if=requirements,
+    )
+
+    result = {
+        'changed': False,
+    }
+
+    client = module.client('polly')
+    s3_client = module.client('s3')
+
+    params = {}
+    if module.params.get('lexicons'):
+        params['LexiconNames'] = module.params.get('lexicons')
+    params['OutputFormat'] = module.params.get('output_format')
+    if module.params.get('sample_rate'):
+        params['SampleRate'] = module.params.get('sample_rate')
+    if module.params.get('speech_mark_types'):
+        params['SpeechMarkTypes'] = module.params.get('speech_mark_types')
+    if module.params.get('text_type'):
+        params['TextType'] = module.params.get('text_type')
+    params['VoiceId'] = module.params.get('voice_id')
+
+    if module.params.get('source') == 'local':
+        params['Text'] = module.params.get('src_path')
+    if module.params.get('source') == 's3':
+        try:
+            response = s3_client.get_object(
+                Bucket=module.params.get('src_bucket'),
+                Key=module.params.get('src_path')
+            )
+            params['Text'] = response['Body'].read().decode("utf-8") 
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e, msg="Failed to get text file from S3")
+
+    create_synthesis(client, s3_client, module, params, result)
```

Comment:
It looks like changed=True is always the case. You don't need to use/pass around a result dict. `module.exit_json(changed=True)`